spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/stores/ConfigStore.java

```diff
@@ -131,21 +131,32 @@ private boolean watchedKeyValid(String watchedKey) {
      */
     public String[] getLabels() {
         if (!StringUtils.hasText(this.getLabel())) {
-            return EMPTY_LABEL_ONLY;
+            return EMPTY_LABEL;
         }
 
+        // The use of trim makes label= dev,prod and label= dev, prod equal.
         List<String> labels =  Arrays.stream(this.getLabel().split(LABEL_SEPARATOR))
-                .filter(StringUtils::hasText)
-                .map(String::trim)
+                .map(label -> mapLabel(label))
                 .distinct()
                 .collect(Collectors.toList());
-
+        
+        if (this.getLabel().endsWith(",")) {
+            labels.add("\0");
```

Comment:
declare this `\0` as a static final string.

Reply:
I have updated it.